qiskit_ibm_runtime/utils/json.py

```diff
@@ -509,8 +513,8 @@ def object_hook(self, obj: Any) -> Any:
             if obj_type == "to_json":
                 return obj_val
             if obj_type == "NoiseModel":
-                if HAS_AER:
-                    return from_dict(obj_val)
+                if HAS_AER and _AER_VERSION[1] >= 16:
+                    return qiskit_aer.noise.NoiseModel.from_dict(obj_val)
```

Comment:
Should we keep support for serializing `NoiseModel`? It will still work with `aer > 16`